thunder/tests/opinfos.py

```diff
@@ -2288,6 +2289,38 @@ def fmod_sample_input_generator(op, device, dtype, requires_grad, **kwargs):
 )
 elementwise_binary_ops.append(logical_xor_opinfo)
 
+
+def ldexp_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    a = make((4, 4), **kwargs)
+    b = make((4, 4), **kwargs)
+    c = make((4, 1), **kwargs)
+
+    yield SampleInput(a, b)
+    yield SampleInput(a, c)
```

Comment:
I had to create this function seperately instead of using `elementwise_binary_generator` because it includes tests for cases where the input tensors are in different device, and these are error cases in our case here in `ldexp`.

https://github.com/Lightning-AI/lightning-thunder/blob/2d18b7a07fb5e83d138707b7bf97bd2f82da4f2c/thunder/tests/opinfos.py#L2038-L2046